torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2720,181 +2721,182 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
         alignment is created by :func:`_shard_parameters`, which ensures that
         the local optimizer only sees the relevant parameter shard.
         """
-        # First hook callback will see PRE state. If we have multiple params,
-        # then subsequent hook callbacks will see POST state.
-        self._assert_state([TrainingState_.BACKWARD_PRE, TrainingState_.BACKWARD_POST])
-        self.training_state = TrainingState_.BACKWARD_POST
-        if param.grad is None:
-            return
+        with torch.autograd.profiler.record_function("FullyShardedDataParallel._post_backward_hooks"):
```

Comment:
This runs per FlatParameter so ideally there would be something in the string to indicate which (id? index? name?) parameter it is running for. But I can't think of a great way to get this information, so let's just save it for a future PR.